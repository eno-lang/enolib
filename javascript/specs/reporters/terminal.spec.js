const enolib = require('../..');
const { TerminalReporter } = require('../..');

const input = `
> comment
# section

field: value

list:
- item
- item

> comment
- item

## subsection

fieldset:
entry = value

> comment
entry = value
`.trim()

describe('Terminal reporter', () => {
  it('produces colored terminal output', () => {
    const document = enolib.parse(input, { reporter: TerminalReporter });

    const snippet = new document._context.reporter(document._context).reportElement(document._context._document.elements[0]).snippet();

    // Uncomment this to inspect the snippet correctness in a terminal for review
    // console.log(snippet);

    expect(snippet).toMatchSnapshot();
  });
});
