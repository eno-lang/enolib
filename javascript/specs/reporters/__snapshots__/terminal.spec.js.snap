// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Terminal reporter produces colored terminal output 1`] = `
"[1m   Line | Content[0m
      1 | [0m[2m[37m>[37m [37mcomment[37m[0m[0m
 >    2 | [1m[41m[35m#[37m [33msection[37m[0m
 *    3 | [33m[0m
 *    4 | [33m[34mfield[37m[35m:[37m [37mvalue[37m[0m
 *    5 | [33m[0m
 *    6 | [33m[34mlist[37m[35m:[37m[0m
 *    7 | [33m[35m-[37m [37mitem[37m[0m
 *    8 | [33m[35m-[37m [37mitem[37m[0m
 *    9 | [33m[0m
 *   10 | [33m[2m[37m>[37m [37mcomment[37m[0m[0m
 *   11 | [33m[35m-[37m [37mitem[37m[0m
 *   12 | [33m[0m
 *   13 | [33m[35m##[37m [33msubsection[37m[0m
 *   14 | [33m[0m
 *   15 | [33m[34mfieldset[37m[35m:[37m[0m
 *   16 | [33m[34mentry[37m [35m=[37m [37mvalue[37m[0m
 *   17 | [33m[0m
 *   18 | [33m[2m[37m>[37m [37mcomment[37m[0m[0m
 *   19 | [33m[34mentry[37m [35m=[37m [37mvalue[37m[0m"
`;
